ibis/backends/base/sql/alchemy/client.py

```diff
@@ -329,3 +329,105 @@ def _sqla_table_to_expr(self, table):
+        data_obj : pd.Dataframe (Later can also be used for sending in
+        table_expr)
+            data_obj is the dataframe containing data which needs to be
+            inserted to to_table_name
+            (Later when table_expr related implementation is done this
+            variable can be used to take in that data)
+        from_table_name : string, optional
+            name of the table from which data needs to be inserted
+        if_exists : string, default 'append'
+            The values available are: {‘fail’, ‘replace’, ‘append’}
+
+        Raises
+        -------
+        ValueError
+            You must pass either data_obj (pandas Dataframe) or
+            from_table_name (table name to insert data from)
+
+        ValueError
+            Sorry, can't insert from both the data_obj (dataframe)
+            and the from_table_name (table). Please use only one
+            parameter.
+
+        ValueError
+            No operation is being performed. Either the data_obj
+            parameter is not a pandas dataframe or the
+            from_table_name parameter is not of string datatype.
+
+        """
+
+        if data_obj is None and from_table_name is None:
+            raise ValueError(
+                'You must pass either data_obj (pandas Dataframe)'
+                ' or from_table_name (table name to insert data from)'
+            )
+
+        if isinstance(data_obj, pd.DataFrame) and isinstance(
+            from_table_name, str
+        ):
+            raise ValueError(
+                "Sorry, can't insert from both the data_obj (dataframe)"
+                " and the from_table_name (table). Please use only one"
+                " parameter."
+            )
+
+        params = {}
+        if database is None:
+            params['schema'] = self.database_name
+            database = self.database_name
+
+        if isinstance(data_obj, pd.DataFrame):
+            data_obj.to_sql(
+                to_table_name,
+                self.con,
+                index=False,
+                if_exists=if_exists,
+                **params,
+            )
+        elif isinstance(from_table_name, str):
+            to_table_expr = self.table(to_table_name)
+            to_table_schema = to_table_expr.schema()
+
+            to_table = self._table_from_schema(
+                to_table_name,
+                to_table_schema,
+                database=database or self.current_database,
+            )
+
+            from_table_expr = self.table(from_table_name)
+
+            with self.begin() as bind:
+                to_table.create(bind=bind)
+                if from_table_expr is not None:
+                    bind.execute(
+                        to_table.insert().from_select(
+                            list(from_table_expr.columns),
+                            from_table_expr.compile(),
+                        )
+                    )
+        else:
+            raise ValueError(
```

Comment:
Not sure if this should be a TypeError instead, since it's raising mainly when the types don't match.

Reply:
I agree. Must be TypeError.